File: curier-service/src/main/java/com/gokhana/courierservice/configuration/KafkaConsumerConfig.java
package com.gokhana.courierservice.configuration;


import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.annotation.EnableKafka;
import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
import org.springframework.kafka.core.ConsumerFactory;
import org.springframework.kafka.support.converter.StringJsonMessageConverter;


@Configuration
@EnableKafka
public class KafkaConsumerConfig {

    final
    ConsumerFactory<String, String> consumerFactory;

    public KafkaConsumerConfig(ConsumerFactory<String, String> consumerFactory) {
        this.consumerFactory = consumerFactory;
    }

    @Bean
    public ConcurrentKafkaListenerContainerFactory<String, String> kafkaListenerContainerFactory() {
        ConcurrentKafkaListenerContainerFactory<String, String> factory = new ConcurrentKafkaListenerContainerFactory<>();
        factory.setMessageConverter(new StringJsonMessageConverter());
        factory.setConsumerFactory(consumerFactory);
        return factory;
    }
}

File: email-service/src/main/kotlin/com/gokhana/emailservice/config/KafkaConsumerConfig.kt
package com.gokhana.emailservice.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.annotation.EnableKafka
import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory
import org.springframework.kafka.core.ConsumerFactory
import org.springframework.kafka.support.converter.StringJsonMessageConverter

@Configuration
@EnableKafka
class KafkaConsumerConfig(val consumerFactory: ConsumerFactory<String, String>) {

    @Bean
    fun kafkaListenerContainerFactory(): ConcurrentKafkaListenerContainerFactory<String, String> {
        val factory = ConcurrentKafkaListenerContainerFactory<String, String>()
        factory.setMessageConverter(StringJsonMessageConverter())
        factory.consumerFactory = consumerFactory
        return factory
    }
}